.gitignore

```diff
@@ -1,2 +1,5 @@
 !.git*
 target/
+.idea
```

Comment:
I consciously did not add any editor-specific gitignore rules here since I believe that belongs to your `~/.gitignore-global`.  It's a pain to have to add exclude rules for every editor out there in every `.gitignore` file.